evm_loader/program/src/executor.rs

```diff
@@ -207,13 +280,19 @@ impl<'config, B: Backend> Handler for Executor<'config, B> {
     }
 }
 
+#[derive(serde::Serialize, serde::Deserialize, Clone, Copy)]
+pub enum RuntimeImpl {
```

Comment:
Maybe `CreateReason` will be a more convenient name for this enum. And I suggest adding Trx (or Root) value to use in case of root runtime executed for contract specified in the transaction. In this case, you can eliminate Option in runtime_info.

Reply:
enum renamed

Reply:
root element added